### PR TITLE
New actions and cards

### PR DESCRIPTION
**Pull request by Concat17**

### CODE REVIEW

File: src/model/ActionSet.cs [hunk 1]
```diff
@@ -203,6 +203,58 @@ public static void PackActionAndExecute(GameState state, Delegate action, int[]
...
+        var player = state.Players[0].Id != state.CurrentPlayer.Id ? state.Players[0] : state.Players[1];
+        state = PerformRandomActionOnPlayer(state, player, action, targetAmount, power);
+      }
+      else
+      {
+        var player = state.CurrentPlayer;
+        state = PerformRandomActionOnPlayer(state, player, action, targetAmount, power);
+      }
+      return state;
+    };
```
Comment (levabala) [inline src/model/ActionSet.cs]: Like a shortcut
``` csharp
public static SpecifiedAction<int, int, int, int> PerformRandomAction = (state, onSelf, action, targetAmount, power, remainArguments) =>
{
  var player = onSelf == 0 ? 
    (state.Players[0].Id != state.CurrentPlayer.Id ? state.Players[0] : state.Players[1]) : 
    state.CurrentPlayer;
    
  return PerformRandomActionOnPlayer(state, player, action, targetAmount, power);      
};
```

File: src/model/ActionSet.cs [hunk 2]
```diff
@@ -203,6 +203,58 @@ public static void PackActionAndExecute(GameState state, Delegate action, int[]
 
       return state;
     };
+
+    public static SpecifiedAction<int, int, int, int> PerformRandomAction = (state, onSelf, action, targetAmount, power, remainArguments) =>
```
Comment (levabala) [inline src/model/ActionSet.cs]:
```suggestion
    public static SpecifiedAction<int, int, int, int> PerformRandomAction = (state, onSelf, action, actionsAmount, power, remainArguments) =>
```

File: src/model/ActionSet.cs [hunk 3]
```diff
@@ -203,6 +203,58 @@ public static void PackActionAndExecute(GameState state, Delegate action, int[]
...
+    };
+
+    private static GameState PerformRandomActionOnPlayer(GameState state, Player player, int action, int targetAmount, int power)
+    {
+      Random rnd = new Random();
+      if (player.ActiveCards.Count < targetAmount)
+      {
+        throw new ArgumentException("Not enough cards on table for this action");
+      }
+      var cardIndexes = Enumerable.Range(0, 2).ToList();
```
Comment (levabala) [inline src/model/ActionSet.cs]: I misunderstand this logic
Why only Range from 0 to 2?

File: src/model/ActionSet.cs [hunk 4]
```diff
@@ -203,6 +203,58 @@ public static void PackActionAndExecute(GameState state, Delegate action, int[]
...
+
+      return state;
+    };
+
+    public static SpecifiedAction<int, int, int, int> BuffCreature = (state, playerId, cardIndex, attackBuff, hpBuff, remainArguments) =>
+    {
+      var player = state.Players[playerId];
+      var creature = player.ActiveCards[cardIndex];
+      creature.Attack += attackBuff;
+      creature.HP += hpBuff;
```
Comment (levabala) [inline src/model/ActionSet.cs:250]: Maybe we need to split props into `actual` and `base`?
To provide an ability to return base params

File: src/model/ActionSet.cs [hunk 5]
```diff
@@ -203,6 +203,58 @@ public static void PackActionAndExecute(GameState state, Delegate action, int[]
...
+      }
+      else
+      {
+        var player = state.CurrentPlayer;
+        state = PerformRandomActionOnPlayer(state, player, action, targetAmount, power);
+      }
+      return state;
+    };
+
+    private static GameState PerformRandomActionOnPlayer(GameState state, Player player, int action, int targetAmount, int power)
```
Comment (levabala) [inline src/model/ActionSet.cs]: Need renaming
This method performs provided action on a random card
But your naming means that "i need to perform random action on a card"
Suggestion: 
- `PerformRandomAction` -> `PerformActionOnCard`
- `PerformRandomActionOnPlayer` -> `PerformActionOnCardWithPlayer`

File: src/model/CardSet.spec.cs [hunk 6]
```diff
@@ -239,5 +239,275 @@ public void StarfireTest()
 
       Assert.AreEqual(host.state, stateExpected);
     }
+
+    [Test]
+    public void Multi_ShotTest()
```
Comment (levabala) [inline src/model/CardSet.spec.cs]: Why not just `MultiShotTest`

**Review by levabala — CHANGES_REQUESTED** [review]

File: src/model/ActionSet.cs (hunk 4 above)
Comment (Concat17) [reply src/model/ActionSet.cs:250]: About base params. Are these parameters before the buff?

Comment (Concat17) [reply src/model/ActionSet.cs:250]: Where do we need to keep this params?

Comment (levabala) [reply src/model/ActionSet.cs:250]: At card struct?

Comment (Concat17) [reply src/model/ActionSet.cs:250]: Ok, then I'll add new field in the next pull

**Pull request merged**